Adds the conveyor belt placer

Citinited opened the pull request:
This PR adds the conveyor belt placer (CBP), which is a glorified bag for conveyor belts. The purpose of this PR is to make conveyor belt placement much quicker and easier (currently it's rather time-consuming to make even short belts).

With the CBP you can:
- Click on the floor with the CBP to apply a belt there, assuming there are belts inside the CBP.
- Click on the CBP with a lever to link all belts with the lever.
- Click on loose belts with the CBP to pick them up.

The normal CBP can be obtained at any autolathe, and has a capacity of 7 belts. The bluespace CBP has to be researched before you can obtain it, and has a capacity of 50 belts.

**Obligatory picture!**

![image](https://user-images.githubusercontent.com/18459142/40332784-49a4a654-5d4d-11e8-9a42-a967fb18140f.png)

 The bluespace one is animated, the normal one is not.
 
🆑 
add: Adds the conveyor belt placer and bluespace conveyor belt placer, allowing you to much more easily create conveyor belts. The former can be gotten at any autolathe, the latter must be researched first.
add: Use a conveyor belt lever on aforementioned item to link all belts inside the placer with that lever.
/🆑

Fox-McCloud commented:
That sprite doesn't look very good.

The idea is interesting though.

Fox-McCloud commented on code/modules/recycling/belt-placer.dm:
```diff
@@ -0,0 +1,42 @@
+	use_to_pickup = TRUE
+	origin_tech = "engineering=1"
+
+/obj/item/storage/conveyor/bluespace
```
I'd rather not have more infinite storage items, please.

The ore is bag is bad enough.

Citinited replied:
Oh, you might be right. I'll limit the number they can store, would 50 be too much?

taukausanake replied:
50 is how much a plant bag can carry of plant stuff. It should be fine there

Citinited commented:
@Fox-McCloud Yeah, the sprite is very much a placeholder. Problem is that I have no idea what this ovject would look like...

Citinited commented:
I've added better sprites to this (see OP), and limited the number of belts that can be held by the bluespace one.

taukausanake commented:
Could you make the bluespace one look more like the normal one, but maybe the window be the bluespace animation? I feel it should look more alike the normal one and less like a Rapid Part Dispenser

Citinited commented:
I changed how the bluespace one was coloured, and added a red light, so hopefully they are sufficiently different now. OP updated

(I'm preserving this picture for comparison - this is how it used to look)
![image](https://user-images.githubusercontent.com/18459142/40327525-37d46bc4-5d3b-11e8-8afa-8fdff27e0fe8.png)

Crazylemon64 commented on code/modules/recycling/belt-placer.dm:
```diff
@@ -0,0 +1,45 @@
+		if(linked)
+			to_chat(user, "<span class='notice'>All belts in [src] linked with [S].</span>")
+	else
+		..()
```
This should be `return ..()` instead

Crazylemon64 commented on code/modules/recycling/belt-placer.dm:
```diff
@@ -0,0 +1,45 @@
+	max_combined_w_class = 200 //50 belts
+	origin_tech = "engineering=2;bluespace=1"
+
+/obj/item/storage/conveyor/attackby(obj/item/conveyor_switch_construct/S, mob/user, params) //So we can link belts en masse
```
This should be `obj/item` instead of specifically the conveyor switch.

Citinited commented:
@Crazylemon64 All issues addressed
